cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/roller/KR2.java

```diff
@@ -18,54 +17,40 @@ public class KR2 {
 
     private final int numPods;
     final Observer observer;
+
+    final CommandPlanner planner;
     final Processor processor;
     private final List<Context> contexts;
 
     public KR2(int numPods, Observer observer, Processor processor) {
         this.numPods = numPods;
         this.observer = observer;
         this.processor = processor;
+        this.planner = new CommandPlannerImpl();
         this.contexts = IntStream.range(0, numPods).boxed()
                 .map(Context::new)
                 .collect(Collectors.toList());
     }
 
     Future<Void> roll() {
-
-        // TODO still need to support roll-controller last
-
         while (true) {
-            int processed = processOne();
```

Comment:
we could refactor to keep a `processOne` method, I can see it would be useful for testing